dotnetv3/Keyspaces/Scenarios/CassandraWrapper.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using System.CodeDom;
+using System.Net;
+using Cassandra;
+
+namespace KeyspacesScenario;
+
+/// <summary>
+/// Class to perform CRUD methods on a Keyspaces Apache Cassandra database.
+/// </summary>
+public class CassandraWrapper
+{
+    private readonly IConfiguration _configuration;
+    private readonly string _localPathToFile;
+    private const string _certLocation = "https://certs.secureserver.net/repository/sf-class2-root.crt";
+    private const string _certFileName = "sf-class2-root.crt";
+    private readonly X509Certificate2Collection _certCollection;
+    private X509Certificate2 _amazoncert;
+    private Cluster _cluster;
+    // User name and password for the service.
+    private string _userName;
+    private string _pwd;
+
+    public CassandraWrapper()
+    {
+        _configuration = new ConfigurationBuilder()
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("settings.json") // Load test settings from .json file.
+            .AddJsonFile("settings.local.json",
+                true) // Optionally load local settings.
+            .Build();
+
+        _localPathToFile = Path.GetTempPath();
+
+        // Get the Starfield digital certificate and save it locally.
+        var client = new WebClient();
+        client.DownloadFile(_certLocation, $"{_localPathToFile}/{_certFileName}");
+
+        _certCollection = new X509Certificate2Collection();
+        _amazoncert = new X509Certificate2($"{_localPathToFile}/{_certFileName}");
+
+        // Get the user name and password stored in the configuration file.
+        _userName = _configuration["UserName"];
+        _pwd = _configuration["Password"];
+
+        var awsEndpoint = "cassandra.us-east-2.amazonaws.com";
+
+        _cluster = Cluster.Builder()
+            .AddContactPoints(awsEndpoint)
+            .WithPort(9142)
+            .WithAuthProvider(new PlainTextAuthProvider(_userName, _pwd))
```

Comment:
Need to do more research on this.

Reply:
It looks as though the code I'm using is secure since as it downloads the certificate and uses the created IAM user and role that was created specifically to access Keyspaces and Apache Cassandra.

Reply:
Ok. I'm not going to hold up the PR for this, because it does work and it matches some current examples. We can have a separate discussion and check with some other parties and revisit it if needed.